ledger/ledgercore/statedelta.go

```diff
@@ -360,21 +442,14 @@ func (ad AccountDeltas) ModifiedAccounts() []basics.Address {
 
 // MergeAccounts applies other accounts into this StateDelta accounts
 func (ad *AccountDeltas) MergeAccounts(other AccountDeltas) {
-	for new := range other.Accts {
-		addr := other.Accts[new].Addr
-		acct := other.Accts[new].AccountData
-		ad.Upsert(addr, acct)
+	for _, balanceRecord := range other.Accts {
+		ad.Upsert(balanceRecord.Addr, balanceRecord.AccountData)
```

Comment:
This kind of change is inefficient, because it copies `other.Accts[i]` into `balanceRecord` at each step.

I think you could get some of the cleanup, and possibly be even faster than the original, with:
```
	for new := range other.Accts {
        balanceRecord := &other.Accts[new]
		ad.Upsert(balanceRecord.Addr, balanceRecord.AccountData)
```

Reply:
Good point, changed in f39564cb4935359401d37e2f72106a77d17dc492